docs/generator.go

```diff
@@ -21,43 +21,53 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+	// detect if file exists
+	_, err = os.Stat(tmpl)
+	// create file if not exists
+	if os.IsNotExist(err) {
+		var file, err = os.Create(tmpl)
+		if err != nil {
+			log.Fatal(err)
+		}
+		file.Close()
+	}
 	f, err := os.OpenFile(tmpl, os.O_APPEND, 0666)
 	value := gjson.GetBytes(content, "properties")
 	value.ForEach(func(key, value gjson.Result) bool {
-		_, err = f.WriteString("\n# " + strings.Title(key.String()) + "\n" + "This section is of " + "{{.Properties." + strings.Title(key.String()) + ".Type}} type that tells about the basic information of the user and consists of the following sub-sections:\n") //print name of sections
+		_, err = f.WriteString("\n# " + strings.Title(key.String()) + "\nThis section is of " + "{{.Properties." + strings.Title(key.String()) + ".Type}} type that tells about the basic information of the user and consists of the following sub-sections:\n") //print name of sections
```

Comment:
I'm not sure if we have a standard or guideline about how to deal with long lines, but no matter the maximum number of columns we should consider, this is probably above it :D maybe you could break it up with a newline before each `+` ?

Reply:
Let's leave it for now and open an issue to clean-up long lines in the future.